### PR TITLE
Create a book class 

### LINKED ISSUE #9: [2pt] Create a Book class
Create a Book class in a separate .rb file.

- All Book class properties visible in the diagram should be defined and set up in the constructor method.

### PR DESCRIPTION
**Pull request by aliabbani**
- Created Item class in a separate .rb file.

- Implemented method: 
Should reuse can_be_archived?() method
Should change the archived property to true if the result of the can_be_archived?() method is true
Should do nothing if the result of the can_be_archived?() method is false

closes #9

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  def initialize(publisher, cover_state, publish_date)
+    super(publish_date)
+    @publisher = publisher
+    @cover_state = cover_state
+    @archived = false
+  end
+
+  def can_be_archived?
+    super || cover_state == 'bad'
+  end
```
Comment (rpire) [inline book.rb:15]: - [x] According to the UML diagram, the `can_be_archived?` method should be a **private method**. What do you think about making it one?

**Review by rpire — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hello, @aliabbani 👋

You did a great job implementing the `Book` class. Instance variables created from `Book` behave as expected, and their methods are working correctly. 🤓

However, I believe there is room for improvement. Please, check the comments below for more details. 🔎

Cheers!

File: book.rb (hunk 1 above)
Comment (aliabbani) [reply book.rb:15]: You are right @rpire thank you for mentioning it. Solved as you can see here:
image

Comment (rpire) [reply book.rb:15]: Done 👍

**Review by rpire — APPROVED** [review]
## Changes Approved 🙌

Hi again @aliabbani 👋

Nice work. You have implemented the requirements. Please, proceed to merge these changes into the `dev` branch. 🚀

Happy coding!

**Pull request merged**